src/m365/teams/commands/user/user-app-list.ts

```diff
@@ -0,0 +1,114 @@
+import * as chalk from 'chalk';
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import Utils from '../../../../Utils';
+import { GraphItemsListCommand } from '../../../base/GraphItemsListCommand';
+import request from '../../../../request';
+import commands from '../../commands';
+import { UserTeamsApp } from '../../UserTeamsApp';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  userId: string;
+  userName: string;
+}
+
+class TeamsUserAppListCommand extends GraphItemsListCommand<UserTeamsApp> {
+  public get name(): string {
+    return `${commands.TEAMS_USER_APP_LIST}`;
+  }
+
+  public get description(): string {
+    return 'Lists the apps deployed in the personal scope of the specified user';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    var userIdPromise: Promise<{ value: string; }>;
+
+    if(args.options.userName) {
```

Comment:
We could make the code easier to read and maintain by putting userId resolution in a separate method.